datadog/dogshell/monitor.py

```diff
@@ -101,8 +102,14 @@ def _post(cls, args):
                 options = json.loads(args.options)
             except:
                 raise Exception('bad json parameter')
+
+        if args.tags:
+            tags = sorted(set([t.strip() for t in args.tags.split(',') if t]))
```

Comment:
I’d check for truth status with the .strip as well.

Reply:
Bump @bradleyjay

Reply:
@platinummonkey Sorry for the delay, thank you for putting eyes on this 🙇

Sure! So we do this in a couple of places in Dogshell, I got this straight from metrics.py:
https://github.com/DataDog/datadogpy/blob/9bb02535c119a90c60399423795743e763cec6c2/datadog/dogshell/metric.py#L46-L48

When you say check for truth status, can you clarify? The one hole I see is if args.tags is something other than a string, though I'm not sure the argparser allows that. But for safety with .strip(), we could swap line 107 to read:

`if args.tags **and type(args.tags) == str:`**

Otherwise, I think we're ok, but I could be misunderstanding - `if t` will evaluate as False if `t` is an empty string, and `if args.tags` will do the same. Either way, if args.tags simply doesn't exist, we're in trouble anyway. If there are no tags, `t.strip` is happy with any string.

Should this be something we change overall with Dogshell? I can make those changes as well, if so. How would you recommend changing it?

Reply:
```python
tags = sorted(set([t.strip() for t in args.tags.split(',') if t.strip()]))
```

because if `t == " "` then `t.strip()` will yield `t == ""` as an added item to the set. when in reality that should not be added.

Reply:
Good god you're right. And it slips right past a str( ) type check. 

Thanks Cody, adding.